pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -655,6 +655,18 @@ def test_non_iso_strings_with_tz_offset(self):
                                            tzinfo=pytz.FixedOffset(240))] * 2)
         tm.assert_index_equal(result, expected)
 
+    def test_timestamp_utc_true(self):
+        # GH 24415
+        ts = Timestamp(2018, 1, 1)
+        result = to_datetime(ts, utc=True)
+        expected = ts.tz_localize('UTC')
+        assert result == expected
+
+        msg = ('Cannot localize tz-aware Timestamp, use tz_convert for '
+               'conversions')
+        with pytest.raises(TypeError, match=msg):
```

Comment:
Does it make sense for the tz-aware `Timestamp` behavior to be different than the behavior for an equivalent tz-aware `datetime` object?  (not super well versed in the tz related conventions we follow)
```python
In [2]: ts = pd.Timestamp('2018-01-01', tz='US/Pacific')

In [3]: dt = ts.to_pydatetime()

In [4]: dt
Out[4]: datetime.datetime(2018, 1, 1, 0, 0, tzinfo=<DstTzInfo 'US/Pacific' PST-1 day, 16:00:00 STD>)

In [5]: pd.to_datetime(dt, utc=True)
Out[5]: Timestamp('2018-01-01 08:00:00+0000', tz='UTC')
```

Reply:
Hmm that's a fair point. You're right; I think we should be consistent to how we treat `datetime.datetime` objects.